### ISSUE TITLE
Original `time` encoding units not used in Zarr generated by `beam-refactor` branch

### ISSUE THREAD

**Issue by derekocallaghan** [pangeo-forge/pangeo-forge-recipes#465]
While porting the EOOffshore CCMP recipe to use the `beam-refactor` branch, I noticed that the [output Zarr contains non-unique `time` values](https://github.com/pangeo-forge/pangeo-forge-recipes/issues/453#issuecomment-1373744070).

It looks like the original dataset `time` `encoding` `units` aren't retained in `pangeo_forge_recipes.writers._store_data()`:

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/eea8e4452a15dc5aeed86df9eefe2130faaf1036/pangeo_forge_recipes/writers.py#L30

At this point the original `time` encoding is as expected (**i.e. `hours...`**): 

```python
{'chunksizes': (524288,), 'fletcher32': False, 'shuffle': False, 'source': '<fsspec.implementations.local.LocalFileOpener object at 0x7f88c259aac0>', 'original_shape': (4,), 'dtype': dtype('float64'), 'units': 'hours since 1987-01-01 00:00:00'}
```

The `units` are subsequently replaced after this statement, where the use of **`days...`** is resulting in the non-unique values:

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/eea8e4452a15dc5aeed86df9eefe2130faaf1036/pangeo_forge_recipes/writers.py#L31

```python
{'chunksizes': (524288,), 'fletcher32': False, 'shuffle': False, 'source': '<fsspec.implementations.local.LocalFileOpener object at 0x7f88c259aac0>', 'original_shape': (4,), 'dtype': dtype('float64'), 'units': 'days since 1970-01-01 00:00:00', '_ARRAY_DIMENSIONS': ['time'], '_CoordinateAxisType': 'Time', '_Fillvalue': -9999.0, 'axis': 'T', 'calendar': 'proleptic_gregorian', 'delta_t': '0000-00-00 06:00:00', 'long_name': 'Time of analysis', 'standard_name': 'time'}
```

I don't have a fix for this yet, but it looks like the issue is in `pangeo_forge_recipes.aggregation._to_variable()` based on the existing comments:

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/eea8e4452a15dc5aeed86df9eefe2130faaf1036/pangeo_forge_recipes/aggregation.py#L198-L203

**Comment by rabernat** [comment]
This is *such a useful bug report*. Thank you so much! 🙏

**Comment by derekocallaghan** [comment]
I have a potential fix for this almost working locally, I need to resolve some test failures. I'm hoping to create a PR later this week.

**Comment by rabernat** [comment]
Amazing! Yes, PR very welcome!

**Comment by derekocallaghan** [comment]
Confirmed that this is resolved by #471.
